### PR TITLE
Some easy fixes for code climate

### PR DESCRIPTION
**Pull request by tshastry**

### CODE REVIEW

File: test_discovery.py [hunk 1]
```diff
@@ -356,10 +362,7 @@ def test_all(test_file):
     t1 = int(time.time())
 
     total_tests = len(tests)
-    failed_tests = 0
-    total_errors = 0
-    total_char_errors = 0
-    total_characters = 0
+    failed_tests = total_errors = total_char_errors = total_characters = 0
```
Comment (cbrochtrup) [inline test_discovery.py]: :+1:

File: test_discovery.py [hunk 2]
```diff
@@ -548,8 +538,7 @@ def main(discovery_directory, test_file, shutdown=True):
     except Exception:
         logger.exception("Error: Check test file for formatting errors", exc_info=True)
 
-    if shutdown:
-        shutdown_discovery()
+    shutdown_discovery(shutdown)
         
     if not success:
         exit(1)
```
Comment (cbrochtrup) [inline test_discovery.py]: The rest of the exits in this script are `sys.exit(1)`. Consistency is nice.
```suggestion
         sys.exit(1)
```

**Comment by mgoldey** [comment]
Fixes 7 issues per https://codeclimate.com/github/finos/greenkey-discovery-sdk/compare/tejas/fix-code-climate

File: test_discovery.py [hunk 3]
```diff
@@ -133,41 +138,41 @@ def json_dump(data, outfile, directory=None):
...
     Loads and parses the test file
     """
-    with open(test_file) as f:
-        test_file = [_.strip() for _ in f if _.strip() and not _.startswith("#")]
-
-    test_file = remove_comments(test_file)
+    test_file = [_.strip() for _ in open(test_file) if _.strip() and not _.startswith("#")]
     test_file, intent_whitelist, domain_whitelist = find_whitelists(test_file)
-
+
```
Comment (mgoldey) [inline test_discovery.py]:
```suggestion

```

File: launch_discovery.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 from discovery_config import CONTAINER_NAME
 
 
-def launch_discovery(custom_directory=None, type=None, port=None, discovery_config=None, container_name=None):
+def launch_discovery(custom_directory=None, type=None, port=DISCOVERY_PORT, discovery_config=DISCOVERY_CONFIG, container_name=CONTAINER_NAME):
```
Comment (mgoldey) [inline launch_discovery.py]: better use of defaults

File: test_discovery.py [hunk 5]
```diff
@@ -26,9 +26,14 @@
...
-from discovery_config import CONTAINER_NAME
-from discovery_config import TIMEOUT, RETRIES
+from discovery_config import (
+  CONTAINER_NAME,
+  DISCOVERY_HOST,
+  DISCOVERY_PORT,
+  DISCOVERY_SHUTDOWN_SECRET, 
+  RETRIES,
+  TIMEOUT,
+)
```
Comment (mgoldey) [inline test_discovery.py]: Good centralization of imports

File: test_discovery.py [hunk 6]
```diff
@@ -548,11 +537,10 @@ def main(discovery_directory, test_file, shutdown=True):
     except Exception:
         logger.exception("Error: Check test file for formatting errors", exc_info=True)
 
-    if shutdown:
-        shutdown_discovery()
+    shutdown_discovery(shutdown)
         
     if not success:
-        exit(1)
+         sys.exit(1)
```
Comment (tshastry) [inline test_discovery.py]:
```suggestion
        sys.exit(1)
```

**Pull request merged**